pymc/distributions/continuous.py

```diff
@@ -1896,6 +1898,140 @@ def icdf(value, nu, mu, sigma):
         )
 
 
+class SkewStudentTRV(RandomVariable):
+    name = "skewstudentt"
+    ndim_supp = 0
+    ndims_params = [0, 0, 0, 0]
+    dtype = "floatX"
+    _print_name = ("SkewStudentT", "\\operatorname{SkewStudentT}")
+
+    @classmethod
+    def rng_fn(cls, rng, a, b, mu, sigma, size=None) -> np.ndarray:
+        return np.asarray(stats.jf_skew_t.rvs(a=a, b=b, size=size, random_state=rng)) * sigma + mu
+
+
+skewstudentt = SkewStudentTRV()
+
+
+class SkewStudentT(Continuous):
+    r"""
+    Skewed Student's T distribution log-likelihood.
+
+    This follows Jones and Faddy (2003)
+
+    The pdf of this distribution is
+
+    .. math::
+
+        f(t)=f(t ; a, b)=C_{a, b}^{-1}\left\{1+\frac{t}{\left(a+b+t^2\right)^{1 / 2}}\right\}^{a+1 / 2}\left\{1-\frac{t}{\left(a+b+t^2\right)^{1 / 2}}\right\}^{b+1 / 2}
+
+    where
+
+    .. math::
+
+        C_{a, b}=2^{a+b-1} B(a, b)(a+b)^{1 / 2}
+
+
+    ========  =============================================================
+    Support   :math:`x \in [\infty, \infty)`
+    Mean      :math:`E(T)=\frac{(a-b) \sqrt{(a+b)}}{2} \frac{\Gamma\left(a-\frac{1}{2}\right) \Gamma\left(b-\frac{1}{2}\right)}{\Gamma(a) \Gamma(b)}`
+    ========  =============================================================
+
+    Parameters
+    ----------
+    a : tensor_like of float
+        First kurtosis parameter (a > 0).
+    b : tensor_like of float
+        Second kurtosis parameter (b > 0).
+    mu : tensor_like of float
+        Location parameter.
+    sigma : tensor_like of float
+        Scale parameter (sigma > 0). Converges to the standard deviation as a and b
+        become close (only required if lam is not specified). Defaults to 1.
+    lam : tensor_like of float, optional
+        Scale parameter (lam > 0). Converges to the precision as a and b
+        become close (only required if sigma is not specified). Defaults to 1.
+
+    """
+
+    rv_op = skewstudentt
+
+    @classmethod
+    def dist(cls, a, b, *, mu=0, sigma=None, lam=None, **kwargs):
+        a = pt.as_tensor_variable(a)
+        b = pt.as_tensor_variable(b)
+        lam, sigma = get_tau_sigma(tau=lam, sigma=sigma)
+        sigma = pt.as_tensor_variable(sigma)
+
+        return super().dist([a, b, mu, sigma], **kwargs)
+
+    def support_point(rv, size, a, b, mu, sigma):
+        a, b, mu, _ = pt.broadcast_arrays(a, b, mu, sigma)
+        Et = mu + (a - b) * pt.sqrt(a + b) * gammafn(a - 0.5) * gammafn(b - 0.5) / (
+            2 * gammafn(a) * gammafn(b)
+        )
+        if not rv_size_is_none(size):
+            Et = pt.full(size, Et)
+        return Et
+
+    def logp(value, a, b, mu, sigma):
+        _, sigma = get_tau_sigma(sigma=sigma)
+
+        x = (value - mu) / sigma
+
+        a_ = (a + 0.5) * pt.log(1 + x / pt.sqrt(a + b + x**2))
+        b_ = (b + 0.5) * pt.log(1 - x / pt.sqrt(a + b + x**2))
+        c = (
+            (a + b - 1) * pt.log(2)
+            + (pt.gammaln(a) + pt.gammaln(b) - pt.gammaln(a + b))
```

Comment:
nitpick use the betaln function.

Reply:
There is currently no `betaln` in `pytensor.tensor` AFAIK

Reply:
It is under the special module. `pt.special.betaln`